src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,51 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_LENGTH_STORAGE = 10;
+    private K[] keyArray = (K[]) new Object[MAX_LENGTH_STORAGE];
+    private V[] valueArray = (V[]) new Object[MAX_LENGTH_STORAGE];
+    private int size = 0;
+
     @Override
     public void put(K key, V value) {
+        if (containsKey(key)) {
+            valueArray[indexOf(key)] = value;
+        } else {
+            keyArray[size] = key;
+            valueArray[size] = value;
+            size++;
+        }
+
```

Comment:
```suggestion
    public void put(K key, V value) {
        int index = indexOf(key);
        if (index != -1) {
            valueArray[index] = value;
        } else {
            keyArray[size] = key;
            valueArray[size] = value;
            size++;
        }
```
you are iterating 2 times when calling ```contains``` and ```indexOf```